[GeoMechanicsApplication] Add more plots for settlement analyses

avdg81 opened the pull request:
**📝 Description**

- Added settlement history plots for test cases 1, 2, and 4.
- Added vertical stress plots along the left edge of the soil columns for test 1, 2, and 4. These plots show the water pressure distribution (except for test case 1, which encompasses a dry column), the vertical effective stress distribution, and the vertical total stress distribution.
- Added data series for the total stress distribution for test case 3.
- The process that fixes the water pressures above the phreatic line has been modified such it will also fix the water pressures of nodes that are **on** the phreatic line.
- Modified an existing unit test to cover the above change.
- Changed the residual saturation to 0.0, to have a fair comparison with D-Settlement.
- As a result of the above change, some expected results had to be updated. Also some time steps were relocated in time.
- Changed the Y coordinate of one node to make sure it is on the phreatic line.
- Updated the test documentation as well as the plots that are included in the documentation.
- When plots are requested, produce them **before** checking the assertions. When an assertion fails, we would like to have plots to inspect.

avdg81 committed:
Added the settlement history plot for test case 1

Also extended the test case's documentation.

avdg81 committed:
Added plots for test cases 2 and 4

Also extended the documentation of these test cases.

markelov208 commented on applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md:
```diff
@@ -31,3 +31,45 @@ The expected values are based on the analytical solution of a fully saturated co
+
+### High permeability
+
+The following plots show some results obtained with Kratos, and compares them with results from D-Settlement 23.2.1.
```
I would re-write it as The following plots show comparisons of results obtained with Kratos and D-Settlement 23.2.1.

avdg81 replied:
Done.

markelov208 commented on applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/low_permeability/test_case_4_settlement_plot.svg:
It looks like there is a big green point at time of 100 days and no other green and orange symbols on the graph.

avdg81 replied:
I have checked the results obtained with Kratos for several time steps starting from 100 days. I've found that these values are there, even though you don't really see them in the plot. Possibly, this is due to the markers which are drawn on top of each other.

There was one thing that I noticed, and which I found interesting. In stage 4, the vertical total displacement of the mid-side node at the top is lagging behind compared to the adjacent corner nodes. In other words, the settlement along the top edge is not truly uniform. This effect disappears after a few time steps, but I cannot explain why it happened in the first place.

If there is anything you'd like me to change, please let me know. Thanks.

markelov208 replied:
Thank you very much for checking. If these symbols overlap each other, is it possible to use other symbols for each line, for example, circle and delta in addition to the cross?

WPK4FEM replied:
The phenomena that cause the non-uniform displacement on the top are:
- ABC has stress dependent stiffness. The vertical stress is dependent on the distance of the integration points to the top surface. Hence stiffness contribution from elements to nodes vary with how close the integration points are to the top surface. For the triangular mesh this potentially gives a non-uniform horizontal distribution of the vertical stiffness. Also the creep is stress dependent.
- Upon loading the pore water pressure takes part of the load. With water being pressed out of the pores, the skeleton of grains takes over. When that happens the effective stress increases and the horizontal distribution of the stiffness becomes more even.

markelov208 reviewed (commented):
Hi Anne, it is so nice to have graphs in the documents. I have two small comments.

avdg81 committed:
Processed a review suggestion

avdg81 committed:
Extended the stress plots of test case 3

They now also show the total stress distributions.

avdg81 committed:
Extended the stress plots of test case 2

They now also show the total stress distributions.

avdg81 committed:
Extended the stress plots of test case 4

They now also show the total stress distributions.

avdg81 committed:
Extended the stress plots of test case 2

We now also have a plot at t = 0 days.

avdg81 committed:
Extended the stress plots of test case 3

We now also have a plot at t = 0 days.

avdg81 committed:
Extended the stress plots of test case 4

We now also have a plot at t = 0 days.

avdg81 committed:
Added a stress plot of test case 1

It shows the vertical effective stress as well as the vertical total stress at t = 0.1 days.

avdg81 committed:
Updated the settlement history plot of test case 1

avdg81 committed:
Added another stress plot of test case 1

It shows the vertical effective stress as well as the vertical total stress at t = 100.1 days.

avdg81 committed:
Added one more stress plot of test case 1

It shows the vertical effective stress as well as the vertical total stress after 10,000 days.

avdg81 committed:
No longer conditionally plot the vertical total stress

Draw it always.

avdg81 committed:
Removed an unused function parameter

Also added a missing blank line.

avdg81 committed:
Improved the settlement test cases

- The process that fixes the water pressures above the phreatic line now also fixes the water pressures of the nodes _on_ the phreatic line.
- Modified a unit test to cover the case where the node is on the phreatic line.
- Set the residual saturation to 0.0.
- Corrected the Y coordinate of a node of test case 3 that was supposed to be _exactly_ on the phreatic line.
- Create the plots of the test cases prior to checking some results.
- Use a slightly larger tolerance for one of the checks.

avdg81 committed:
Made it clearer that one of the nodes is _on_ the phreatic line

rfaasse committed:
Revalidated test after change in behavior of the fix phreatic line process

WPK4FEM commented on applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md:
```diff
@@ -31,3 +31,55 @@ The expected values are based on the analytical solution of a fully saturated co
+
+### High permeability
+
+The following plots show comparisons of results obtained with Kratos and D-Settlement 23.2.1.
```
I wonder if comparison is enough, should that also be multiple?

avdg81 replied:
I'm sorry, but I don't understand what you mean exactly. Could you please elaborate on that? Thanks.

WPK4FEM replied:
comparisons of results -> comparison of results

WPK4FEM commented on applications/GeoMechanicsApplication/tests/dsettlement/fully_saturated_column_uniform_load/README.md:
```diff
@@ -31,3 +31,55 @@ The expected values are based on the analytical solution of a fully saturated co
+
+![Settlement](high_permeability/test_case_2_settlement_plot.svg)
+
+The following plots show the water pressure distribution ($`p_{\mathrm{w}}`$), the vertical effective stress distribution ($`\sigma_{\mathrm{eff, yy}}`$), and the vertical total stress distribution ($`\sigma_{\mathrm{tot, yy}}`$) along the left edge of the soil column at certain times.
```
The texts repeat for high and low permeability. There may be less duplication if it is announced once, before the ###High permeability caption

avdg81 replied:
Yes, it's a good habit to reduce duplication as much as we can. Thanks for pointing this out. Done.

WPK4FEM reviewed (dismissed):
Dear Anne,
This became quite a large pull request, but it gave a lot more insight in the stress fields through depth. Reading through I think that still some reuse of functionality could be achieved. However, I think it is important to safeguard this work now by merging. We can always revisit when further tests are being documented in a similar way.
Thank you,
Wijtze Pieter

avdg81 committed:
Updated the test documentation as suggested by Wijtze Pieter

avdg81 committed:
Merge remote-tracking branch 'refs/remotes/origin/geo/add-more-plots-for-settlement' into geo/add-more-plots-for-settlement

avdg81 committed:
Processed one more review comment

markelov208 reviewed (approved):
Hi Anne, the very impressive perfect match between D-Settlement and Kratos solutions. The PR is good to be merged. Hope @WPK4FEM could explain me why RESIDUAL_SATURATION has to be zero.
